### PR TITLE
feat: use nVersion=2 for all transactions to reduce wallet fingerpinting

### PR DESCRIPTION
**Pull request by peter-sanderson**
Fixies: https://github.com/trezor/trezor-suite/issues/15374

Produce transactions with `nVersion=2` to reduce a fingerprint surface as it seems most of the wallets use the version 2. And it enables the [BIP68](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki)

Sources:
- https://ishaana.com/blog/wallet_fingerprinting/
   > A basic transaction has version number of 1, but most modern bitcoin transactions now use a version number of 2.
  
- https://learnmeabitcoin.com/technical/transaction/#structure-version
   > Only nVersion 1 and 2 are standard. Use of certain opcodes requires nVersion to be set to 2. Some wallets generally use version 2, while others still use version 1.
   
   
   
image

### CODE REVIEW

File: packages/connect/e2e/tests/device/unlockPath.test.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ describe('TrezorConnect.unlockPath', () => {
 
         expect(unlockedSignTx.payload).toMatchObject({
             serializedTx:
-                '010000000001010ab6ad3ba09261cfb4fa1d3680cb19332a8fe4d9de9ea89aa565bd83a2c082f90100000000ffffffff02c8736e0000000000225120c5c7c63798b59dc16e97d916011e99da5799d1b3dd81c2f2e93392477417e71e50c30000000000001976a914a579388225827d9f2fe9014add644487808c695d88ac014006bc29900d39570fca291c038551817430965ac6aa26f286483559e692a14a82cfaf8e57610eae12a5af05ee1e9600acb31de4757349c0e3066701aa78f65d2a00000000',
```
Comment (peter-sanderson) [inline packages/connect/e2e/tests/device/unlockPath.test.ts:200]: `01...` -> `02...`

File: packages/connect/src/data/coinInfo.ts [hunk 2]
```diff
@@ -174,11 +174,14 @@ export const getCoinName = (path: number[]) => {
...
 
+const BITCOIN_SHORTCUTS = ['BTC', 'TEST', 'REGTEST'];
+
 const parseBitcoinNetworksJson = (json: any) => {
     Object.keys(json).forEach(key => {
         const coin = json[key];
         const shortcut = coin.coin_shortcut;
-        const isBitcoin = shortcut === 'BTC' || shortcut === 'TEST';
+
+        const isBitcoin = BITCOIN_SHORTCUTS.includes(shortcut);
```
Comment (mroz22) [inline packages/connect/src/data/coinInfo.ts:184]: nitpick, isn't this a standalone commit actually?

**Review by mroz22 — APPROVED** [review]
looks okeyish. not sure if anybody else is interested, maybe @marekrjpolak ?

File: packages/connect/e2e/tests/device/unlockPath.test.ts (hunk 1 above)
Comment (peter-sanderson) [reply packages/connect/e2e/tests/device/unlockPath.test.ts:200]: I had to change it. after this segment `0000000001010ab6ad3ba09261cfb4fa1d3680cb19332a8fe4d9de9ea89aa565bd83a2c082f90100000000ffffffff02c8736e0000000000225120c5c7c63798b59dc16e97d916011e99da5799d1b3dd81c2f2e93392477417e71e50c30000000000001976a914a579388225827d9f2fe9014add644487808c695d88ac0140` its different

**Comment by marekrjpolak** [comment]
@mroz22 We've discussed it with @peter-sanderson yesterday and from my side it's ok. I just wanted him to test even the Big Beautiful Collaborative Transaction™ before we merge it.

File: packages/connect/src/data/coinInfo.ts (hunk 2 above)
Comment (peter-sanderson) [reply packages/connect/src/data/coinInfo.ts:184]: https://github.com/trezor/trezor-suite/pull/15373/commits/c712e10bc44ee1ae5f1c51ccacfb420b9aaeb6de

**Pull request merged**